src/m365/spo/commands/folder/folder-add.ts

```diff
@@ -45,14 +56,35 @@ class SpoFolderAddCommand extends SpoCommand {
       },
       {
         option: '-n, --name <name>'
+      },
+      {
+        option: '--color [color]'
       }
     );
   }
 
   #initValidators(): void {
     this.validators.push(
-      async (args: CommandArgs) => validation.isValidSharePointUrl(args.options.webUrl)
-    );
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.color !== undefined) {
+          if (typeof args.options.color === "number") {
```

Comment:
Currently, we're checking if it's either a number or a string. From the past, we've learned that it's not a good idea to let minimist decide which type a value is.
Therefore I suggest that:
- We parse the option color always as a string.
- Validate the option against an array of possible values by using `Object.entries(...).flat()`. Note that we'll have to type the numbers as string in the mapper object.